muonDIS/makeMuonDIS.py

```diff
@@ -0,0 +1,204 @@
+
+    muon_vetoPoints    = r.TClonesArray("vetoPoint")
+    dTree.Branch('muon_vetoPoints', muon_vetoPoints, 32000, -1)
+
+    myPythia = r.TPythia6()
+    myPythia.SetMSEL(2)
+    myPythia.SetPARP(2, 2)
+    for kf in [211, 321, 130, 310, 3112, 3122, 3222, 3312, 3322, 3334]:
+        kc = myPythia.Pycomp(kf)
+        myPythia.SetMDCY(kc, 1, 0)
+
+    R = int(time.time() % 900000000)
+    myPythia.SetMRPY(1, R)
+    mutype = {-13: 'gamma/mu+', 13: 'gamma/mu-'}
+
+    myPythia.SetMSTU(11, 11)
+    print(f"Processing events from {nStart} to {nEnd}...")
+
+    nMade = 0
+    
+    print(nStart, nEnd)
+    
+    for k in range(nStart, nEnd):
+        
+        rc = sTree.GetEvent(k)
+
+        imuondata = sTree.imuondata
+        
+        pid = imuondata[0]
+        px = imuondata[1]
+        py = imuondata[2]
+        pz = imuondata[3]
+        x = imuondata[4]
+        y = imuondata[5]
+        z = imuondata[6]
+        w = imuondata[7]
+        time_muon=imuondata[8]
+        
+        p = r.TMath.Sqrt(px ** 2 + py ** 2 + pz ** 2)
+        E = r.TMath.Sqrt(getMasssq(pid) + p ** 2)
+        theta = r.TMath.ACos(pz / p)
+        phi = r.TMath.ATan2(py, px)
+        ctheta, stheta = r.TMath.Cos(theta), r.TMath.Sin(theta)
+        cphi, sphi = r.TMath.Cos(phi), r.TMath.Sin(phi)
+
+        #print(f"Muon: PID={pid}, px={px}, py={py}, pz={pz}, E={E}, x={x}, y={y}, z={z}, w={w}")
+        
+        isProton = 1
+        xsec = 0
+
+        mu = array('d', [pid, px, py, pz, E, x, y, z, w, isProton, xsec,time_muon])
+        muPart = r.TVectorD(12, mu)
+        myPythia.Initialize('FIXT', mutype[pid], 'p+', p)
+        myPythia.Pylist(1)
+        target='p+'
+        
+        for a in range(args.nDIS):
+            if a == args.nDIS // 2:
+                myPythia.Initialize('FIXT', mutype[pid], 'n0', p)
+                isProton = 0
+                target='n0'
+                print(f"Switching to neutron interaction")
+
+            dPartDIS.Clear()
+            iMuon.Clear()
+            muPart[9] = isProton
+            iMuon[0] = muPart
+            myPythia.GenerateEvent()
+            myPythia.Pyedit(1)
+            print(f"Event {a} generated, number of particles: {myPythia.GetN()}")
+
+            
+            for itrk in range(1, myPythia.GetN() + 1):
+                xsec = myPythia.GetPARI(1)
+                muPart[10] = xsec
+                did = myPythia.GetK(itrk, 2)
+                dpx, dpy, dpz = rotate(ctheta, stheta, cphi, sphi, myPythia.GetP(itrk, 1), myPythia.GetP(itrk, 2), myPythia.GetP(itrk, 3))
+                psq = dpx ** 2 + dpy ** 2 + dpz ** 2
+                E = r.TMath.Sqrt(getMasssq(did) + psq)
+                m = array('d', [did, dpx, dpy, dpz, E])
+                part = r.TVectorD(5, m)
+                nPart = dPartDIS.GetEntries()
+                if dPartDIS.GetSize() == nPart:
+                    dPartDIS.Expand(nPart + 10)
+                dPartDIS[nPart] = part
```

Comment:
This should be adapted for ROOT 6.32 as done in https://github.com/ShipSoft/FairShip/pull/559